### PR TITLE
Adding the changes for vagrant ethernet public network

### PR DESCRIPTION
**Pull request by raghusarangapani**
The vagrant file under public_network assumes en0 wifi as the connection type. It does not work for Thunderbolt ethernet connections. This change adds support to Thunderbolt ethernet connection.

generate_cluster_configs_from_vagrant_hosts.py was also modified to accommodate this new vagrant file.

### CODE REVIEW

File: utilities/generate_cluster_configs_from_vagrant_hosts.py [hunk 1]
```diff
@@ -10,26 +10,28 @@
...
+def generate_cluster_configs_from_vagrant(private_network, public_network, public_network_ethernet):
     """
     1. Gets the status for a running vagrant vm set.
     2. Uses the host name to look up the ip allocated to each vagrant vm instance
     3. Uses this IP list to build a pool.json file and generate the cluster configurations
     """
 
-    if private_network and public_network:
-        raise ProvisioningError("Invalid private_network and public_network flags")
+    if private_network and public_network and public_network_ethernet:
```
Comment (sethrosetter) [inline utilities/generate_cluster_configs_from_vagrant_hosts.py]: Add a different type of verification. What were are looking for is that only one flag is set and the other two are not. You could use a count to make sure only one is set or use a series of asserts.

File: vagrant/public_network_ethernet/Vagrantfile [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# configures the configuration version (we support older styles for
+# backwards compatibility). Please don't change it unless you know what
+# you're doing.
+Vagrant.configure(2) do |config|
+
+  host_ips = [
+    "host1",
+    "host2",
+    "host3",
+    "host4"
```
Comment (sethrosetter) [inline vagrant/public_network_ethernet/Vagrantfile]: Add a host5. You can see that this is the setting by default on master

File: utilities/generate_cluster_configs_from_vagrant_hosts.py (hunk 1 above)
Comment (raghusarangapani) [inline utilities/generate_cluster_configs_from_vagrant_hosts.py]: Added an else clause to keep it simple

File: vagrant/public_network_ethernet/Vagrantfile (hunk 2 above)
Comment (raghusarangapani) [inline vagrant/public_network_ethernet/Vagrantfile]: Added

File: utilities/generate_cluster_configs_from_vagrant_hosts.py [hunk 3]
```diff
@@ -19,17 +19,21 @@ def generate_cluster_configs_from_vagrant(private_network, public_network):
 
     if private_network and public_network:
         raise ProvisioningError("Invalid private_network and public_network flags")
+    elif private_network and public_network_ethernet:
+        raise ProvisioningError("Invalid private_network and public_network_ethernet flags")
```
Comment (sethrosetter) [inline utilities/generate_cluster_configs_from_vagrant_hosts.py]: Would public_network and public_network_ethernet still fall through?

I would check if one flag is set, the other two are not.

Comment (raghusarangapani) [inline utilities/generate_cluster_configs_from_vagrant_hosts.py]: Fixed

**Comment by sethrosetter** [comment]
Can you run `flake8` from the root of the repo and fix the following errors?
```
./utilities/generate_cluster_configs_from_vagrant_hosts.py:36:1: E101 indentation contains mixed spaces and tabs
./utilities/generate_cluster_configs_from_vagrant_hosts.py:36:1: W191 indentation contains tabs
./utilities/generate_cluster_configs_from_vagrant_hosts.py:38:1: E101 indentation contains mixed spaces and tabs
./utilities/generate_cluster_configs_from_vagrant_hosts.py:96:1: W293 blank line contains whitespace
```

File: utilities/generate_cluster_configs_from_vagrant_hosts.py [hunk 4]
```diff
@@ -10,26 +10,22 @@
...
     2. Uses the host name to look up the ip allocated to each vagrant vm instance
     3. Uses this IP list to build a pool.json file and generate the cluster configurations
     """
 
-    if private_network and public_network:
-        raise ProvisioningError("Invalid private_network and public_network flags")
-
-    if not private_network and not public_network:
-        raise ProvisioningError("Invalid private_network and public_network flags")
-
```
Comment (sethrosetter) [inline utilities/generate_cluster_configs_from_vagrant_hosts.py]: Why did you move this from the method into main? If someone was importing / calling this from another script, this validation would not longer be run.
